truffle/src/main/java/org/jruby/truffle/nodes/core/RegexpNodes.java

```diff
@@ -126,6 +126,13 @@ public Object match(RubyRegexp regexp, RubyString string) {
             return regexp.matchCommon(string, true, false) != getContext().getCoreLibrary().getNilObject();
         }
 
+        @Specialization
+        public Object match(RubyRegexp regexp, RubySymbol symbol) {
+            notDesignedForCompilation();
+
+            return regexp.matchCommon(symbol.toRubyString(), true, false) != getContext().getCoreLibrary().getNilObject();
```

Comment:
In these kinds of situations we probably want to factor out `matchCommon` to use `ByteList`, rather than converting `Symbol` to `String`.